Convert One Time payment button to Hyperlink

tahaf22labs opened the pull request:
Ticket: https://app.asana.com/0/0/1203046498775533/f

- To make the monthly button visually prominent
Notes:
- I've updated the button to look like Hyperlink instead of replacing it with Anchor Tag directly. So that I won't be missing important attributes such as "name", "disabled" which are not present in Anchor Tag
- ~~Here, I've created a separate button - DonateLink since the DonateButton uses a custom Button.~~
Update: I've updated it to use Donation Button as suggested by @yeseniamolinab 


Update: Adding Screenshots
- Desktop
//One-Time Payment
<img width="1768" alt="image" src="https://user-images.githubusercontent.com/105354588/193355170-4fb43b69-8400-4af2-8f87-ffe530630abf.png">
//Recurring
<img width="1792" alt="image" src="https://user-images.githubusercontent.com/105354588/193355247-ca8eb01e-4444-4f47-8a0b-4e0d23914ae7.png">

-Mobile
<img width="468" alt="image" src="https://user-images.githubusercontent.com/105354588/193355301-7f6c8eb8-0b8c-456b-8efa-a22a85ea5a71.png">

yeseniamolinab commented on app/javascript/components/DonateLink.js:
```diff
@@ -0,0 +1,56 @@
+      onClick={props.disabled ? null : props.onClick}
+    >
+      <ProcessingThen processing={props.submitting || false}>
+        {'Or'}
```
```suggestion

```

tahaf22labs replied:
> @tahaf22labs You can simplify this; instead of creating a new component that is almost an exact copy of the previous one to make it look different, you can modify the existing one to match the requirements.
> 
> Since you are using a button and only modifying the styling, then you can reuse the `DonateButton` component and modify the `Button` component to now receive an extra property; take this as an example:
> 
> ```js
> const Button = ({ onClick, className, disabled, theme, type, name }) => {
>     const classes = classNames(styles.button, styles[theme], className);
>     // theme being the new property that will help you define what styles to use
>     // styles.button being the common styles between the two (if any)
> ```
> 
> Then when you're using it:
> 
> For the monthly Button:
> 
> ```
> <DonateButton
>         currency={currency}
>         amount={donationAmount || 0}
>         submitting={this.state.submitting}
>         name="recurring"
>         recurring={true}
>         recurringDonor={this.state.recurringDonor}
>         weekly={this.props.weekly}
>         disabled={this.disableSubmit()}
>         onClick={e => this.onClickHandle(e)}
>         theme="primary"
>  />
> ```
> 
> For the one-off link:
> 
> ```
> <DonateButton
>         currency={currency}
>         amount={donationAmount || 0}
>         submitting={this.state.submitting}
>         name="one_time"
>         recurring={false}
>         recurringDonor={this.state.recurringDonor}
>         disabled={this.disableSubmit()}
>         onClick={e => this.onClickHandle(e)}
>         theme="secondary"
>  />
> ```
> 
> And finally, on the Button component:
> 
> ```
> < Button
>       className="DonateButton"
>       name={props.name}
>       onClick={props.onClick}
>       disabled={props.disabled}
>       theme={props.theme}
>     >
> ```
> 
> Then on the `Button.scss` you will have two different themes (primary being the styles of the regular Button and secondary being the styles for the link).
> 
> Could you please add screenshots to the PR description? (covering how the change looks on desktop and mobile view).
> 
> Please let me know if you have any questions.

Thank you for your precise feedback! I've updated the CSS, Can you please have a look and let me know if that fits!

yeseniamolinab reviewed (changes requested):
@tahaf22labs You can simplify this; instead of creating a new component that is almost an exact copy of the previous one to make it look different, you can modify the existing one to match the requirements.

Since you are using a button and only modifying the styling, then you can reuse the `DonateButton` component and modify the `Button` component to now receive an extra property; take this as an example:

```javascript
const Button = ({ onClick, className, disabled, theme, type, name }) => {
    const classes = classNames(styles.button, styles[theme], className);
    // theme being the new property that will help you define what styles to use
    // styles.button being the common styles between the two (if any)
```

Then when you're using it:

For the monthly Button:
```html
<DonateButton
        currency={currency}
        amount={donationAmount || 0}
        submitting={this.state.submitting}
        name="recurring"
        recurring={true}
        recurringDonor={this.state.recurringDonor}
        weekly={this.props.weekly}
        disabled={this.disableSubmit()}
        onClick={e => this.onClickHandle(e)}
        theme="primary"
 />
```

For the one-off link:
```html
<DonateButton
        currency={currency}
        amount={donationAmount || 0}
        submitting={this.state.submitting}
        name="one_time"
        recurring={false}
        recurringDonor={this.state.recurringDonor}
        disabled={this.disableSubmit()}
        onClick={e => this.onClickHandle(e)}
        theme="secondary"
 />
```

And finally, on the Button component:
````html
< Button
      className="DonateButton"
      name={props.name}
      onClick={props.onClick}
      disabled={props.disabled}
      theme={props.theme}
    >
````

Then on the `Button.scss` you will have two different themes (primary being the styles of the regular Button and secondary being the styles for the link).

Could you please add screenshots to the PR description? (covering how the change looks on desktop and mobile view).

Please let me know if you have any questions.

yeseniamolinab commented:
@tahaf22labs, Thank you so much for addressing the feedback.
I will ask you about another change you weren't aware of before, and I just found out after a meeting.
This change must happen only when we have the double buttons on the UI.

That means that, if the URL contains `recurring=only_one_off` only the one-off button will show, and in that scenario, we still want the regular styling.

I recommend you to use `this.state.recurringDefault` to verify which theme you need to use for the one-off button.

```javascript
theme={this.state.recurringDefault=== 'only_one_off' ? "primary" ? "secondary"}
```

Something like that

tahaf22labs commented:
> ```js
> only_one_off
> ```

Somehow, Even I was thinking this could be a possibility. To check this out, I created a few campaigns under the Fundraiser template. But none of those options displayed a single one_time button. Hence didn't consider this edge case.
Post you pointed this out, I found there's some weird logic error in the Code. Where, We are considering "only_one_off" to display the button but there's no such case where we'll have "only_one_off" but "one_off"
<img width="694" alt="Screenshot 2022-09-30 at 2 06 10 PM" src="https://user-images.githubusercontent.com/105354588/193228806-15a74045-3809-46b1-a168-466434642e47.png">

Let's have a call to have a discussion on this @yeseniamolinab .

tahaf22labs commented:
> @tahaf22labs, Thank you so much for addressing the feedback. I will ask you about another change you weren't aware of before, and I just found out after a meeting. This change must happen only when we have the double buttons on the UI.
> 
> That means that, if the URL contains `recurring=only_one_off` only the one-off button will show, and in that scenario, we still want the regular styling.
> 
> I recommend you to use `this.state.recurringDefault` to verify which theme you need to use for the one-off button.
> 
> ```js
> theme={this.state.recurringDefault=== 'only_one_off' ? "primary" ? "secondary"}
> ```
> 
> Something like that

Somehow, Even I was thinking this could be a possibility. To check this out, I created a few campaigns under the Fundraiser template. But none of those options displayed a single one_time button. Hence didn't consider this edge case.
Post you pointed this out, I found there's some weird logic error in the Code. Where, We are considering "only_one_off" to display the button but there's no such case where we'll have "only_one_off" but "one_off"
<img width="694" alt="Screenshot 2022-09-30 at 2 06 10 PM" src="https://user-images.githubusercontent.com/105354588/193228806-15a74045-3809-46b1-a168-466434642e47.png">

Let's have a call to have a discussion on this @yeseniamolinab .

yeseniamolinab reviewed (approved):
LGTM 💯
